Update the sample code for ebplane daemon to use the new cc_ebpf target.

ccontavalli-j opened the pull request:
See how the example code has changed. We should now be ready to write code to load and install the ebpf code in the kernel.

mhines99 commented on build/embed.bzl:
```diff
@@ -143,19 +158,20 @@ def _cc_embed(ctx):
       accessors.append("inline const std::string_view {var}(&{start}, &{end} - &{start});".format(**subs))
 
-    ifguard = _clean_path(hfile.path).upper()
+    ifguard = _clean_path(hfile.path).upper() + "_"
```
add a comment here on what ifguard means - i am assuming this is a "header define guard"
?

ccontavalli-j replied:
done

anandrao79 commented on lib/ebpf/BUILD:
```diff
@@ -0,0 +1,10 @@
+package(default_visibility = ["//lib:__subpackages__"])
+
```
Since there can be 2 types of ebpf code, one which runs in kernel mode (restricted C code) and another which is userspace ebpf code, it is better to name the directory/file accordingly with _kern or _user suffix.

ccontavalli-j replied:
Discussed over slack. The ebpf/ directory only contains .c code that is compiled into eBPF to run in restricted mode in the kernel (eg, the _kern code). Management code will be C++ in the parent directory (eg, the _user code_).

I have added a README indicating the purpose of the ebpf/ directory. Given the difference in language and build rules, the different directories, and the README, we were both ok not adding further suffixes.

ccontavalli-j commented:
I have rebased the change, now the diff is clear. Anand, I have also addressed the comment above as per chat over slack.

Could you have another look? 

Note that tests will not pass until we fix the toolchain and cloud environment. Still working on that.

jscherpelz-juniper commented on build/embed.bzl:
```diff
@@ -66,6 +66,21 @@ def _clean_path(directory):
+   
+   Args:
+     s: string, generally a filename.
+     extensions: list of strings, extensions to remove.
```
nit: You might want to mention that the order of extensions is important. If I'm reading this correctly you must supply them in reverse order relative to 's'

ccontavalli-j replied:
good suggestion, removed.

jscherpelz-juniper commented on build/embed.bzl:
```diff
@@ -231,11 +248,19 @@ def _cc_embed(ctx):
            doc = "Targets to turn into strings and embed in the binary."
        ),
+       "namespace": attr.string(
+           default = "embedded",
```
nit: I think users should always(?) specify a namespace. Is it possible to make the default more obvious, for example "unspecified_namespace"

ccontavalli-j replied:
well, I think the common use for this rule is to embed static data, eg, a .txt file, a .gif, ...

I can mark this attribute as mandatory, and force the user of the rule to always specify a namespace, or assume that the common case is to embed some data - a handful of files - and it's ok to have a default namespace?

I'd leave as is unless you feel strongly.
